### ISSUE TITLE
PolyATrimmer option USE_NEW_TRIMMER causes NullPointerException

### ISSUE THREAD

**Issue by cljacobs** [broadinstitute/Drop-seq#137]
### Affected tool(s)
PolyATrimmer

### Affected version(s)
- [x] Latest public release version [2.3.0]
- [ ] Latest development/master branch as of [date of test?]

### Description 
I'm getting a NullPointerException whenever I try to call PolyATrimmer with the option USE_NEW_TRIMMER=true (as per the cookbook). I've attached a log from one of the files I tried running, but there's scant little information in it beyond the exception. The outputs do appear to think the trimmer has run, the final line (before the exception) is the summary line (time elapsed and memory used information). However, the time is 0.00 minutes (sometimes 0.01) and the resulting files are empty.

Running without the USE_NEW_TRIMMER option does not cause this issue.

### Logs
[1.log](https://github.com/broadinstitute/Drop-seq/files/3462205/trim-polya.1.log)
[2.log](https://github.com/broadinstitute/Drop-seq/files/3462207/trim-polya.1.log)

**Comment by alecw** [comment]
Hi @cljacobs ,

Have you tagged molecular barcodes as per the cookbook?  Note that by default the new trimmer searches for adapter sequence that is a combination of a fixed adapter string and reverse-complement of contents of XM tag (molecular barcode).  If your reads don't have XM tag, you'll get this error.

Regards, Alec

**Comment by cljacobs** [comment]
Hi @alecw,

I've corrected the molecular barcode tags (see below), but I'm still getting the same null pointer exception that runs the command but ultimately produces an empty BAM whenever I try to use the new trimmer.

I had originally tagged the reads with their molecular barcodes using the "OX" tag rather than the default "XM" tag. I had modified/added appropriate options to the other commands to specify this tag (e.g. in `DetectBeadSynthesisErrors` and `DigitalExpression`), but the help for `PolyATrimmer` only mentions the "ZP" tag. I tried supplying the MOLECULAR_BARCODE_TAG option used by the other commands anyway, but as suspected, it's not a recognized option.

**Comment by alecw** [comment]
Hi @cljacobs ,

Just to make sure I understand: your reads do not have an XM tag, but instead have OX tag, is that right?  If that is the case, then you need to tell PolyATrimmer about this, as follows:

`ADAPTER=~OX~XCACGTACTCTGCGTTGCTACCACTG`

This means: when looking for adapter sequence to trim, use the reverse complement of OX tag value, concatenated with reverse complement of XC value, concatenated with the bases that follow.

If adding the above argument doesn't work, send a handful of reads from your input BAM and I'll take a look.

Regards, Alec

**Comment by cljacobs** [comment]
Hi @alecw,

I ended up with two files. The original file had "OX" tag. I also re-ran the tagger (`TagBamWithReadSequenceExtended`) on an upstream file, so that I had a file with the "XM" tag. Other than the name of the tag these files were identical. Both of them were generating the error, but both have now been resolved. Thanks very much for helping me out with this.

For the original file, changing the ADAPTER argument works as expected.

The error I was seeing in the "XM" file was due to a typo I had made in linking the file. Fixing the typo also fixed the error I was seeing with that file. Sorry about the confusion this caused. It ran, gave identical output to the above file, and there wasn't any need to change the ADAPTER argument, since I was using the default tags, anyway.

Thanks again.
